draft-ietf-webtrans-http2.md

```diff
@@ -203,10 +203,11 @@ of !SEMANTICS=I-D.ietf-httpbis-semantics}}. The WebTransport server MUST verify
 the `Origin` header to ensure that the specified origin is allowed to access
 the server in question.
 
-From the client's perspective, a WebTransport session is established when the
-client receives a 200 response. From the server's perspective, a session is
-established once it sends a 200 response. Both endpoints MUST NOT send any
-WebTransport capsules on a given session before that session is established.
+The client MAY optimistically send any WebTransport capsules that are allowed by
+flow control, for example to send datagrams or flow control updates, for a
+session for which it has sent the CONNECT request, even if it has not yet
+received the server's response to the request. On the server side, this is
+possible as soon as the CONNECT request has been received.
```

Comment:
A session is established once the server sends a 200 (or 2xx ?) response.  A server generates a 2xx response on the request header, not its content.  A server MUST NOT process datagrams or capsules unless it accepts the WebTransport session.  A client can therefore preemptively send WebTransport capsules or datagrams associated with a CONNECT request without waiting for a response, to the extent allowed by the connection (which includes flow control).  This can reduce latency for any data sent by a client at the start of a WebTransport session.  It also ensures that a client can resend the same data when re-attempting to establish a session that was rejected by a server.

Note that this latter one is an interesting thing.  It means that we can easily support various 4xx features easily, including those that allow for automatic retry, such as 425 and 421 and maybe some of those error statuses that include Retry-After.

Reply:
Very nice, thank you! I reworked the paragraph a bit to take most of that phrasing into account, additional tweaks welcome :)